internal/brokers/manager_test.go

```diff
@@ -0,0 +1,209 @@
+			if tc.wantErr {
+				require.Error(t, err, "BrokerForSessionID should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "BrokerForSessionID should not return an error, but did")
+
+			require.Equal(t, tc.wantBroker, got.Name, "BrokerForSessionID should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestGetBroker(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		brokerID string
+
+		wantBroker string
+		wantErr    bool
+	}{
+		"Successfully returns expected broker": {brokerID: "local", wantBroker: "local"},
+
+		"Error when broker does not exist": {brokerID: "does not exist", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			got, err := m.GetBroker(tc.brokerID)
+			if tc.wantErr {
+				require.Error(t, err, "GetBroker should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "GetBroker should not return an error, but did")
+			require.Equal(t, tc.wantBroker, got.ID, "GetBroker should return the expected broker, but did not")
+		})
+	}
+}
+
+func TestSetDefaultBrokerForUser(t *testing.T) {
+	t.Parallel()
+
+	m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+	require.NoError(t, err, "Setup: could not create manager")
+
+	brokers := m.AvailableBrokers()
+	//#nosec // This just to vary the selected broker in the tests, we don't care about randomness safety.
+	i := rand.Intn(len(brokers))
+
+	want, err := m.GetBroker(brokers[i].ID)
+	require.NoError(t, err, "Setup: could not get broker")
+
+	m.SetDefaultBrokerForUser("user", want)
+
+	got := m.BrokerForUser("user")
+	require.Equal(t, want.ID, got.ID, "BrokerForUser should return the expected broker, but did not")
+}
+
+func TestManagerEndSession(t *testing.T) {
+	t.Parallel()
+
+	tests := map[string]struct {
+		sessionID string
+
+		wantErr bool
+	}{
+		"Successfully end existing session": {sessionID: "success"},
+
+		"Error when broker does not exist":      {sessionID: "does not exist", wantErr: true},
+		"Error when broker can not end session": {sessionID: "ES_error", wantErr: true},
+	}
+	for name, tc := range tests {
+		tc := tc
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			b := newBrokerForTests(t)
+
+			m, err := brokers.NewManager(context.Background(), nil, brokers.WithRootDir(rootDir))
+			require.NoError(t, err, "Setup: could not create manager")
+
+			if tc.sessionID != "does not exist" {
+				m.SetBrokerForSessionID(tc.sessionID, &b)
+			}
+
+			err = m.EndSession(context.Background(), tc.sessionID)
+			if tc.wantErr {
+				require.Error(t, err, "EndSession should return an error, but did not")
+				return
+			}
+			require.NoError(t, err, "EndSession should not return an error, but did")
```

Comment:
As discussed, let’s try to get back a broker from sessionID here and see it’s not available anymore.